src/sqlfluff/rules/L007.py

```diff
@@ -60,49 +67,149 @@ def _eval(self, context: RuleContext) -> LintResult:
+                for s in memory["since_code"]:
+                    if s.name == "newline":
+                        anchor = context.segment
+                        # Had a newline - so mark this operator as a fail
+
+        elif memory["last_code"] and memory["last_code"].is_type(
+            "binary_operator", "comparison_operator"
+        ):
+            # It's not an operator, but the last code was.
+            if operator_new_lines != "before":
+                # If in "after" mode, then check to see
+                # there is a newline between us and the last operator.
+                for s in memory["since_code"]:
+                    if s.name == "newline":
+                        # Had a newline - so mark last operator as a fail
+                        anchor = memory["last_code"]
+
+        # Prepare memory for later
+        memory["last_code"] = context.segment
+        memory["since_code"] = []
 
         # Anchor is our signal as to whether there's a problem
-        if anchor:
-            return LintResult(anchor=anchor, memory=memory, description=description)
-        else:
+        if not anchor:
             return LintResult(memory=memory)
+
+        expr = cast(Tuple[BaseSegment], parent_stack[-1].segments)
+        fixes = _generate_fixes(expr, anchor, operator_new_lines)
+        return LintResult(
+            anchor=anchor,
+            memory=memory,
+            description=description,
+            fixes=fixes,
+        )
+
+
+def _generate_fixes(
+    expr: Tuple[BaseSegment],
+    anchor: BaseSegment,
+    operator_new_lines: str,
+) -> List[LintFix]:
+    """Generate Fixes.
+
+    Take operator_new_lines==after as the default case.
+    Consider "before" as a modification on this behaviour.
+
+    """
+    fixes = [LintFix.delete(anchor)]
+    res = list(_get_surrounding_segments(expr, anchor))
+    if operator_new_lines == "before":
+        res = [list(reversed(els)) for els in reversed(res)]
+
+    change_list, anchor_list = res
+    insert_anchor = anchor_list[-1]
+    change_list = change_list[1:]
+    for el in change_list:
+        fixes.append(LintFix.delete(el))
+
+    change_list.append(anchor)
+    if operator_new_lines == "before":
+        # We do yet another reverse here,
+        # This could be avoided but makes all "changes" relate to "before" config state
+        change_list = list(reversed(change_list))
+
+    change_list = list(map(_copy_el, reversed(change_list)))
+    edit_type = "create_after" if operator_new_lines == "before" else "create_before"
+    fixes.append(
+        LintFix(
+            edit_type=edit_type,
+            edit=change_list,
+            anchor=insert_anchor,
+        )
+    )
+    return fixes
+
+
+def _copy_el(segment: BaseSegment):
+    """Safely duplicate segment without position markers."""
+    el = copy.deepcopy(segment)
+    el.pos_marker = None  # type: ignore
+    return el
+
+
+def _get_surrounding_segments(
+    expr: Tuple[BaseSegment],
```

Comment:
Does `expr` contain multiple `BaseSegments`? If so, I think this should be:```suggestion
    expr: Tuple[BaseSegment, ...],
```